src/coreclr/tools/Common/TypeSystem/IL/Stubs/RuntimeHelpersIntrinsics.cs

```diff
@@ -75,13 +75,24 @@ public static MethodIL EmitIL(MethodDesc method)
                         result = true;
                         break;
                     default:
-                        var mdType = elementType as MetadataType;
-                        if (mdType != null && mdType.Name == "Rune" && mdType.Namespace == "System.Text")
-                            result = true;
-                        else if (mdType != null && mdType.Name == "Char8" && mdType.Namespace == "System")
-                            result = true;
-                        else
-                            result = false;
+                        result = false;
+                        if (elementType is MetadataType mdType)
+                        {
+                            if (mdType.Module == mdType.TypeSystemContext.SystemModule &&
+                                mdType.Namespace == "System.Text" &&
+                                mdType.Name == "Rune" && )
```

Comment:
```suggestion
                                mdType.Name == "Rune")
```

Reply:
Ugh. Thanks. I need to get this compiling locally; still having trouble even with main.